internal/controller/apiextensions/composite/composition_ptf_test.go

```diff
@@ -1111,6 +1111,114 @@ func TestFunctionIODesired(t *testing.T) {
 	}
 }
 
+func TestWithKubernetesAuthentication(t *testing.T) {
+	errBoom := errors.New("boom")
+
+	type params struct {
+		c              client.Reader
+		namespace      string
+		serviceAccount string
+	}
+	type args struct {
+		ctx context.Context
+		fn  *v1.ContainerFunction
+		r   *fnv1alpha1.RunFunctionRequest
+	}
+	type want struct {
+		r   *fnv1alpha1.RunFunctionRequest
+		err error
+	}
+
+	cases := map[string]struct {
+		reason string
+		params params
+		args   args
+		want   want
+	}{
+		"GetServiceAccountError": {
```

Comment:
`WithKubernetesAuthentication` can return 5 different error types, but we test only for two here. Could we add tests for the rest of them?

Reply:
I considered it when I was writing this test. I'm inclined not to. Triggering the three errors we're not covering right now would require us to either:

* "Test through" to the underlying implementations, e.g. by providing a reference that causes `name.ParseReference` to fail.
* Refactor the code under test to make it possible to inject all the go-containerregistry dependencies (like `name.ParseReference`) with fake implementations.

Generally I do prefer to get full coverage, but in this case I feel the code complexity required to make all these dependencies injectable doesn't really warrant the increased test coverage.